### ISSUE TITLE
When doing sign with CashID callback URL in incognito mode that URL is called from non-incognito context

### ISSUE THREAD

**Issue by tbrannt** [BadgerWallet/badger#13]
I integrated badger as a login-method for my service. Authentication via badger works fine in normal mode but in chrome incognito mode it doesn't. When I activate badger in incognito mode and try to login the session of the incognito window will not get the session cookie set but the one of the non-incognito browser window.

**Comment by SpendBCH** [comment]
Are you saying the CashID request is being sent from the normal window's account instead of from the incognito window's account?

Otherwise, can you verify that each window has a different session ID to be signed?

Thank you! @GitCash 1000 bits for @tbrannt

**Comment by GitCash** [comment]
Hey **tbrannt**, user **SpendBCH** tipped you **1000 bits** in Bitcoin Cash ( ~ **$0.155** ).

*[Click here to claim it!](https://gitcash.io/claim/tbrannt)*

You can also add the "thumbs down" reaction to **SpendBCH**'s comment above to prevent future tips.

**Comment by tbrannt** [comment]
> Are you saying the CashID request is being sent from the normal window's account instead of from the incognito window's account?

No I used the same badgerwallet accounts in both windows, the incognito one and the normal one.

So what I did is the following: I opened a normal and an incognito window. In both I have the badger extension unlocked, activated and the same account chosen. When I now navigate to the login page in the incognito browser window and sign the cashid message then the cashid callback URL will be called from the non-incognito context resulting in the site setting the sessionId cookie in the non-incognito browser window. 

So even though I signed in in my incognito window I'm now logged in in the non-incognito window while the incognito browser is still logged out.

As I'm in the badger telegram group as I think you also are I'll send you a link to my application there. It's not really public yet so I'll rather not post it here yet.
